src/codeu/chat/server/Server.java

```diff
@@ -116,6 +124,10 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         Serializers.INTEGER.write(out, NetworkCode.NEW_CONVERSATION_RESPONSE);
         Serializers.nullable(ConversationHeader.SERIALIZER).write(out, conversation);
+      if (type == NetworkCode.SERVER_INFO_REQUEST) {
```

Comment:
See my above comment - you only need to make that change once, so you can probably delete this section.